controllers/cloudstackfailuredomain_controller.go

```diff
@@ -130,21 +140,222 @@ func (r *CloudStackFailureDomainReconciliationRunner) ClearMachines() (ctrl.Resu
 	if err := r.K8sClient.List(r.RequestCtx, machines, client.MatchingLabels{infrav1.FailureDomainLabelName: r.ReconciliationSubject.Name}); err != nil {
 		return ctrl.Result{}, err
 	}
-	// Deleted CAPI machines for CloudStack machines found.
-	for _, machine := range machines.Items {
+	r.Log.Info(fmt.Sprintf("Clear machine: %d found.", len(machines.Items)))
+	if len(machines.Items) > 0 {
+		if result, err := checkClusterReady(r); err != nil {
+			return result, err
+		}
+	}
+
+	result, err := triggerEtcdClusterRollout(machines.Items, r)
+	if err != nil {
+		return result, err
+	}
+
+	result, err = triggerControlPlaneRollout(machines.Items, r)
+	if err != nil {
+		return result, err
+	}
+
+	result, err = triggerMachineDeploymentRollout(machines.Items, r)
+	if err != nil {
+		return result, err
+	}
+
+	if len(machines.Items) > 0 {
+		return r.RequeueWithMessage("FailureDomain still has machine(s) in it.")
```

Comment:
I think this message could use more detail, in isolation of other context. Would it be more accurate/meaningful to say "Deleted FailureDomain still has machine(s) in it"?

Reply:
`return r.RequeueWithMessage("failuredomain delete: FailureDomain still has machine(s) in it.", "failuredomain", r.ReconciliationSubject.Spec.Name)`

Reply:
what about these machines makes them noteworthy? I would say 
```
return r.RequeueWithMessage("Deleted FailureDomain still has machine(s) in it which need to be cleared. Requeueing", "failuredomain", r.ReconciliationSubject.Spec.Name)
```